packages/cinterop/src/commonMain/kotlin/io/realm/kotlin/internal/interop/UUIDWrapper.kt

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2022 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm.kotlin.internal.interop
+
+/**
+ * Wrapper around Core UUID values.
+ * See https://github.com/realm/realm-core/blob/master/src/realm/uuid.hpp for more information
+ */
+interface UUIDWrapper {
```

Comment:
Any specific reason to have a full interface here and not just a typealias.

Reply:
A type alias would not be distinguishable from a regular `ByteArray`, as the type system resolves them as the same type.

Reply:
I forgot that we need this to differentiate the incoming values 👍